pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -319,24 +319,37 @@ def __init__(self, param, depth):
         self.optsChanged(self.param, self.param.opts)
 
     def updateDepth(self, depth):
-        ## Change item's appearance based on its depth in the tree
-        ## This allows highest-level groups to be displayed more prominently.
-        if depth == 0:
-            for c in [0,1]:
-                self.setBackground(c, QtGui.QBrush(QtGui.QColor(100,100,100)))
-                self.setForeground(c, QtGui.QBrush(QtGui.QColor(220,220,255)))
-                font = self.font(c)
-                font.setBold(True)
+        """
+        Change item's appearance based on its depth in the tree
+        This allows highest-level groups to be displayed more prominently.
+        """
+        app = QtWidgets.QApplication.instance()
+        palette = app.palette()
+        textColor = palette.text().color()
+        backgroundColor = palette.button().color().darker(110)
+        altBackgroundColor = palette.dark().color()
+
+        # if stylesheet and 'QDarkStyleSheet' in stylesheet:
+        if app.styleSheet(): # Assume the stylesheet is qdarkstyle
+            # #455364 is the background-color of the QHeaderView, #E0E1E3 is lineEdit text from QDarkStyle
+            # lighter/darker are applied to the colors to maintain a good contrast with the item:hover color
+            backgroundColor = QtGui.QColor('#455364').darker(100)
+            altBackgroundColor =  backgroundColor.lighter(130)
+            textColor = QtGui.QColor('#E0E1E3')
```

Comment:
Can we get around hardcoding colors here, in the local code where it cannot be overridden?
It seems to me that that is what caused the original issue: The hardcoded values that you patched out in L326.

If I see it correctly, this is supposed to address one specific issue:
When running on windows, qdarkstyle is the best way to realize a dark mode color scheme. The problem? qdarkstyle individually skins all elements by stylesheet, and never bothers to set up a matching QPalette. The widgets provided by pyqtgraph are not in that stylesheet, so they fall back to the palette and appear basically color swapped.

So the problem in that case is the missing palette.

Why don't we fix *that* instead?
Since we already rely on mkApp to set 'darkMode' for us, can't we just sanity check the palette there?
E.g. if the stylesheet is qdarkstyle, but the base color is white... Then we manually change the application palette to a dark one, like I posted above.

Yes, it's still hardcoded. But the user can override it, and that would automatically change the appearance of GroupParameterItem, too. Even if it is done at runtime.

Reply:
From what I remember choosing a color palette in the windows settings really only affects the Microsoft apps(maybe native applications too) so Qt apps would not pick up these changes in the palette. 

The only problem I see for doing this in `mkQApp` is the users who use their own `QApplication` won't get see the changes. If I can get a conformation that this is okay, i think this would be a good way to handle this, although like you said, this would change the entire application palette, not just the parameterTree.

This would be a better option than applying the palette directly to the parameterTree because the window stays gray but the parameterTree is dark blue, which looks a bit weird.

Reply:
I was just thinking about this and realized that we aren't going to be able to check for a stylesheet here. `mkQApp` creates the QApplication and one needs to exist before a stylesheet can be applied to it.

Reply:
Hmm. Good point.
Since the stylesheet gets set on the QApplication itself, it seems that Qt did not bother to include an event callback; other widgets have a `changeEvent()` method that should be called for this change.

That is a pity. 
A) We could give mkQApp a parameter to pass in style sheets, then the code can handle that correctly.
  Then add some documentation why mkQApp is great.
B) We could figure out if there is a signal for a style change that CAN be accessed even when we only have the QApplication
C) We could subclass QApplication with something that looks intercepts and handles the `setStyleSheet()` call.
D) We could monkey-patch QAPP to replace the setStyleSheet call with our own; which does stuff

Ideas? (A) might be reasonably safe...

Reply:
Option A does sound nice compared to what I've done

I'm not aware of a way to override something like `setStyleSheet()` without a subclass and when I was looking through the change log I noticed we had QApplication sub-classed but decided to change it to this function because QApplication is already a singleton. There was also a comment in `Qt/__init__.py` about segfaults being caused by a subclass of QApplication on PySide2.6/Python 3.8.7.

> Since the stylesheet gets set on the QApplication itself, it seems that Qt did not bother to include an event callback; other widgets have a `changeEvent()` method that should be called for this change.
actually this is kind of what I went with.

I figured since `QApplication` inherits from `QObject` I could just patch the `eventFIlter` and listen for `QEvent.StyleChange`

This is what I ended up with: https://gist.github.com/Wubbzi/50eea2eceae6d4220a0003ae5815fab9

For some reason  `QEvent.ApplicationPaletteChange` triggers `QEvent.StyleChange`, so If someone applies their own palette and the onPaletteChanged runs again, sees that the base color isn't pure white, their palette will be overridden with the darkPalette. 

Right now, using a global as a flag to keep onPaletteChanged from running more than once is the best way I can think of to solve this. 

Let me know what you think.